wapitiCore/attack/mod_methods.py

```diff
@@ -39,6 +39,48 @@ class ModuleMethods(Attack):
     do_post = True
     excluded_path = set()
 
+    async def test_compare_methods(self, request: Request, methods: set):
+        response_dict = {}
+        # posting a GET request to compare
+        # with the uncommon methods
+        try:
+            log_verbose(f"[+] {request}")
+            get_response = await self.crawler.async_send(request)
+        except RequestError:
+            self.network_errors += 1
+            return
+        # Testing and comparing the uncommon methods,
+        # not_allowed_methods record all the methods
+        # returning a 405 error which will be removed
+        not_allowed_methods = set()
+        for method in methods:
+            uncommon_request = Request(
```

Comment:
is it important to keep specify that the request is uncommon ? 

Naming it _request_ should be clear enough.

Reply:
request is already refering to an argument of the function. In fact, it is the OPTIONS request that is passed in argument, I could rename it to free the name "request" but i will have to rename the argument variable into something like "option_request", I guess it will just shift the problem around.